src/SplitPackages/Program.cs

```diff
@@ -5,26 +5,86 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.Extensions.Logging;
 
 namespace SplitPackages
 {
     class Program
     {
-        static void Main(string[] args)
+        private const int Ok = 0;
+        private const int Error = 1;
+
+        private ILogger _logger;
+        private bool _whatIf;
+
+        private CommandLineApplication _app;
+        private CommandOption _sourceOption;
+        private CommandOption _csvOption;
+        private CommandOption _destinationOption;
+
+        public Program(
+            CommandLineApplication app,
+            CommandOption sourceOption,
+            CommandOption csvOption,
+            CommandOption destinationOption,
+            CommandOption whatIfOption)
         {
-            if (args.Contains("help", StringComparer.OrdinalIgnoreCase) ||
-                args.Length < 3 ||
-                args.Length > 4)
-            {
-                PrintUsage();
-                return;
-            }
+            _app = app;
+            _sourceOption = sourceOption;
+            _csvOption = csvOption;
+            _destinationOption = destinationOption;
+            _whatIf = whatIfOption.HasValue();
+
+            InitializeLogger();
+        }
+
+        static int Main(string[] args)
+        {
+            var app = new CommandLineApplication();
+            app.Name = "SplitPackages";
+
+            app.HelpOption("-?|-h|--help");
 
-            bool whatif = args.Length == 4 && args[3].Equals("whatif", StringComparison.OrdinalIgnoreCase);
+            var sourceOption = app.Option(
+                    "--source <DIR>",
+                    "The directory containing the nuget packages to split in different folders",
+                    CommandOptionType.SingleValue);
 
+            var csvOption = app.Option(
+                "--csv <PATH>",
+                "Path to the CSV file containing the names and subfolders where each package must go",
+                CommandOptionType.SingleValue);
+
+            var destinationOption = app.Option(
+                "--destination <DIR>",
+                "The directory on which subdirectories will be created for the different destinations of the packages",
+                CommandOptionType.SingleValue);
+
+            var whatIfOption = app.Option(
+                "--whatif",
+                "Performs a dry run of the command with the given arguments without copying the packages",
+                CommandOptionType.NoValue);
+
+            var program = new Program(app, sourceOption, csvOption, destinationOption, whatIfOption);
+
+            app.OnExecute(new Func<int>(program.Execute));
```

Comment:
The compiler is not smart enough to determine the right overload to use, so I need to rely on the new Func<...>